### PR TITLE
Implement scenery remove game actions.

### PR DESCRIPTION
**Pull request by ZehMatt**
Implemented ClearAction, SceneryRemoveSmallAction and SceneryRemoveLargeAction. With this we have the first group of commands complete.

```
GAME_COMMAND_CLEAR_SCENERY
-> GAME_COMMAND_REMOVE_PATH
-> GAME_COMMAND_REMOVE_SCENERY
-> GAME_COMMAND_REMOVE_WALL
-> GAME_COMMAND_REMOVE_LARGE_SCENERY
```

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Perhaps #7381 could be reused to an extent…?

**Comment by ZehMatt** [comment]
Reused how? Is something wrong?

**Comment by Gymnasiast** [comment]
I think that Aaron was referring to this:

> Only GAME_COMMAND_CLEAR_SCENERY is left as the root command which should be fairly simple to do later.

**Comment by ZehMatt** [comment]
I've added @IntelOrca changes from his previous PR, now the whole group is complete.

**Comment by AaronVanGeffen** [comment]
> I've added @IntelOrca changes from his previous PR, now the whole group is complete.

Mind setting the author for the commit in question to @IntelOrca? :)

**Comment by ZehMatt** [comment]
Done

**Comment by ZehMatt** [comment]
@Gymnasiast renamed them, can you take another peek?

File: src/openrct2-ui/title/TitleSequencePlayer.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ class TitleSequencePlayer final : public ITitleSequencePlayer
 
     int32_t _lastScreenWidth = 0;
     int32_t _lastScreenHeight = 0;
-    CoordsXY _viewCentreLocation = {};
+    CoordsXY _viewCentreLocation;
```
Comment (Gymnasiast) [inline src/openrct2-ui/title/TitleSequencePlayer.cpp]: Seems to be unrelated.

File: src/openrct2/paint/VirtualFloor.cpp [hunk 2]
```diff
@@ -290,10 +290,10 @@ static void virtual_floor_get_tile_properties(
 void virtual_floor_paint(paint_session* session)
 {
     static constexpr const CoordsXY scenery_half_tile_offsets[4] = {
-        { -32, 0 },
-        { 0, 32 },
-        { 32, 0 },
-        { 0, -32 },
+        CoordsXY{ -32, 0 },
```
Comment (Gymnasiast) [inline src/openrct2/paint/VirtualFloor.cpp]: Is this needed? It doesn't look like it.

**Comment by Gymnasiast** [comment]
I have left two line notes, otherwise the code looks good. But I haven't tested it yet.

File: src/openrct2/actions/LargeSceneryRemoveAction.hpp [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+            if (!(gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) && !gCheatsSandboxMode)
+            {
+                if (!map_is_location_owned(currentTile.x, currentTile.y, currentTile.z))
+                {
+                    return MakeResult(GA_ERROR::NO_CLEARANCE, STR_CANT_REMOVE_THIS, STR_LAND_NOT_OWNED_BY_PARK);
+                }
+            }
+
+            // NOTE: Not sure what to make of this, is this actually correct?
```
Comment (duncanspumpkin) [inline src/openrct2/actions/LargeSceneryRemoveAction.hpp]: This is tied to `ResetClearLargeSceneryFlag`. It prevents a double+ cost when the clear scenery tool overlaps multiple tiles of a large scenery element. I've changed the comment in my quarter tile refactor. #8722
